### PR TITLE
refactor: download function

### PR DESCRIPTION
**Pull request by Yelinz**
refresh when downloadURL is expired
extract download function from document-card

This fix prevents serving expired download and webdav urls. A fresh webdav link is always fetched before opening as we dont know when it expires. The download url is parsed for expiry date and only refetched if its expired.

Also fixes the problem that after a replace the old file is still served.

### CODE REVIEW

**Comment by czosel** [comment]
conflicts :-/

File: addon/helpers/download.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import { DateTime } from "luxon";
+
+export function isDownloadUrlExpired(url) {
+  const expiryRegex = /expires=([0-9]+)/;
+  const match = url.match(expiryRegex);
+  return DateTime.fromSeconds(Number(match[1])) < DateTime.now();
+}
```
Comment (czosel) [inline addon/helpers/download.js:7]: isn't the `helpers` folder usually reserved for template helpers? maybe use "utils" instead?

**Comment by czosel** [comment]
LGTM for the rest, nice work! :100:

File: addon/helpers/download.js (hunk 1 above)
Comment (Yelinz) [reply addon/helpers/download.js]: Done moved utils into utils

Comment (Yelinz) [reply addon/helpers/download.js]: Done

**Review by czosel — APPROVED** [review]

**Pull request merged**